src/crypt_prov.rs

```diff
@@ -203,11 +203,62 @@ impl<'a> ImportOptions<'a> {
             }
         }
     }
+
+    /// Imports a DER-encoded PKCS8 private key.
+    pub fn import_pkcs8(&mut self, der: &[u8]) -> io::Result<CryptKey> {
+        unsafe {
+            assert!(der.len() <= winapi::DWORD::max_value() as usize);
+
+            // Decode the der format into a CRYPT_PRIVATE_KEY_INFO struct
+            let mut buf = ptr::null_mut();
+            let mut len = 0;
+            let res = wincrypt::CryptDecodeObjectEx(wincrypt::X509_ASN_ENCODING |
+                                                    wincrypt::PKCS_7_ASN_ENCODING,
+                                                    wincrypt::PKCS_PRIVATE_KEY_INFO,
+                                                    der.as_ptr(),
+                                                    der.len() as winapi::DWORD,
+                                                    wincrypt::CRYPT_DECODE_ALLOC_FLAG,
+                                                    ptr::null_mut(),
+                                                    &mut buf as *mut _ as winapi::LPVOID,
+                                                    &mut len);
+            if res == winapi::FALSE {
+                return Err(io::Error::last_os_error());
+            }
+            let pkey: wincrypt::CRYPT_PRIVATE_KEY_INFO = *buf;
+
+            // Decode pkey's internal der blob again into the desired DSS V3 Private Key BLOB
```

Comment:
Can this be implemented by calling `self.import`?

Reply:
Yes it can, done